src/main/scala/edu/ie3/simona/service/primary/PrimaryServiceWorker.scala

```diff
@@ -78,15 +80,30 @@ final case class PrimaryServiceWorker[V <: Value](
         Try {
           /* Set up source and acquire information */
           val factory = new TimeBasedSimpleValueFactory(valueClass, timePattern)
-          val source = new CsvTimeSeriesSource(
-            csvSep,
-            directoryPath,
-            fileNamingStrategy,
-            timeSeriesUuid,
-            filePath,
-            valueClass,
-            factory
-          )
+          val source = Try(
+            new CsvTimeSeriesSource(
+              csvSep,
+              directoryPath,
+              fileNamingStrategy,
+              timeSeriesUuid,
+              filePath,
+              valueClass,
+              factory
+            )
+          ) match {
```

Comment:
This error matching would also be nice for SqlTimeSeriesSource as well, no? So maybe we can move this an abstraction level upwards, i.e. checking for `Failure` after *TimeSeriesSource has been created.